### PR TITLE
[FIX](complextype)fix struct nested complex collection type and and regresstest

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
this pr do 
1. fix array/map/struct write next_rowid for num_rows when we read from segmentfile
2.improve struct can nested complex type for load action
3.add one/two-level regress test for complex types

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run pipelinex_p0

**Comment by amorynan** [comment]
run p1

File: regression-test/suites/datatype_p0/nested_types/base_cases/one_level_nestedtypes_with_s3data.groovy [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    String sk = getS3SK()
+    String s3_endpoint = getS3Endpoint()
+    String bucket = context.config.otherConfigs.get("s3BucketName");
+
+
+    def dataFilePath = "https://"+"${bucket}"+"."+"${s3_endpoint}"+"/regression/datalake"
+//    def dataFilePath = "/mnt/disk1/wangqiannan/export/ol"
+    def table_names = ["test_array_one_level", "test_map_one_level", "test_struct_one_level"]
+
+    def colNameArr = ["c_bool", "c_tinyint", "c_smallint", "c_int", "c_bigint", "c_largeint", "c_float",
```
Comment (xiaokang) [inline regression-test/suites/datatype_p0/nested_types/base_cases/one_level_nestedtypes_with_s3data.groovy:36]: colNames

File: be/src/vec/data_types/serde/data_type_struct_serde.cpp [hunk 2]
```diff
@@ -94,106 +93,125 @@ Status DataTypeStructSerDe::deserialize_one_cell_from_json(IColumn& column, Slic
...
-            if (!ret.second) {
-                return Status::InvalidArgument("Struct field name {} is duplicate with others.",
-                                               field_names[i]);
+        } else if (c == '\\' && idx + 1 < slice_size) { //escaped
+            ++idx;
+        } else if (!has_quote && (c == '[' || c == '{')) {
+            ++nested_level;
+        } else if (!has_quote && (c == ']' || c == '}')) {
+            --nested_level;
+        } else if (!has_quote && nested_level == 0 && c == options.map_key_delim && !key_added) {
```
Comment (xiaokang) [inline be/src/vec/data_types/serde/data_type_struct_serde.cpp:129]: what's map_key_delim?

**Review by xiaokang — COMMENTED** [review]
can you separate set_num_rows fix, struct deserialize_one_cell_from_json and testcase to different prs?

File: be/src/vec/data_types/serde/data_type_struct_serde.cpp (hunk 2 above)
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_struct_serde.cpp:129]: ","

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run pipelinex_p0

**Comment by amorynan** [comment]
run pipelinex_p0

**Review by morningman — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**